galley/pkg/kube/source/init.go

```diff
@@ -15,18 +15,86 @@
 package source
 
 import (
+	"io/ioutil"
 	"fmt"
+	"os"
 	"time"
+	"strings"
 
+	"github.com/ghodss/yaml"
 	"github.com/hashicorp/go-multierror"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
-
 	"istio.io/istio/galley/pkg/kube"
 	kube_meta "istio.io/istio/galley/pkg/metadata/kube"
 )
 
+func writeCRDs(cs clientset.Interface, fileName string) (error) {
+	CRDData, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		scope.Infof("Read CRD %v", err)
+		return err
+	}
+
+	// yaml.YAMLtoJSON does not appear to understand "---".
+	yamlStrings := strings.Split(string(CRDData), "---")
+	// Build a set of yaml strings and range over them.
+	for _, yamlStringCRD := range yamlStrings {
+		var yamlCRD apiextensionsv1beta1.CustomResourceDefinition
+		yaml.Unmarshal([]byte(yamlStringCRD), &yamlCRD)
+
+		_, err = cs.ApiextensionsV1beta1().CustomResourceDefinitions().Create(&yamlCRD)
+		if err != nil {
+			scope.Infof("Error writing CRDs: %v", err)
```

Comment:
I am worried about the error being ignored here. The logic should be resilient to dealing with existing resources, but we should also be able to cleanly fail in case when we cannot set the custom resource definition correctly.

Reply:
ya me too.  The implementation can use a little/lot of work :)  I wanted to get this up for review before further progress proceeds on other work to see if this approach is even viable.

Reply:
I think the overall approach is fine. But we should be clear about what the plan is:
- 1.1: Galley installs CRDs as part of first phase of two-phase installation.
- 1.2: Galley stops doing CRD installation. It is done by the operator instead. Galley is installed in phase two.

This is the plan, right?